Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/layout/ThreadLayout.java

```diff
@@ -443,6 +444,46 @@ public void highlightPostTripcode(String tripcode) {
         threadListLayout.highlightPostTripcode(tripcode);
     }
 
+    @Override
+    public void filterPostSubject(String subject) {
+        callback.openFilterForType(FilterType.SUBJECT, subject);
+    }
+
+    @Override
+    public void filterPostName(String name) {
+        callback.openFilterForType(FilterType.NAME, name);
+    }
+
+    @Override
+    public void filterPostID(String id) {
+        callback.openFilterForType(FilterType.ID, id);
+    }
+
+    @Override
+    public void filterPostComment(CharSequence comment) {
+        callback.openFilterForType(FilterType.COMMENT, comment.toString());
+    }
+
+    @Override
+    public void filterPostCountryCode(Post post) {
+        String countryCode = "";
+        if (post.httpIcons != null && !post.httpIcons.isEmpty()) {
+            for (PostHttpIcon icon : post.httpIcons) {
+                if (icon.url.toString().contains("troll") || icon.url.toString().contains("country")) {
```

Comment:
From strings.xml, 
`For country code filtering, troll country codes should be prepended with <tt>t_</tt>. A country code filter automatically implies the case insensitive flag for regex.`
You should probably add that `t_` prefix in here for troll codes, as they're processed different than regular country codes.

Reply:
The country code is taken from the icon name so it already has the "t_" prefix for troll codes.
https://github.com/Adamantcheese/Kuroba/blob/1bd1adeda736e233262c2dd89c6b6a76705bf4bc/Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/common/FutabaChanReader.java#L281-L283